edx_django_utils/ip/internal/ip.py

```diff
@@ -0,0 +1,285 @@
+"""
+Implementation of ``edx_django_utils.ip`` -- contains non-public
+functions, and is subject to breaking changes without warning.
+
+Import from ``edx_django_utils.ip`` instead of this one.
+See that module's docstring for additional background.
+"""
+
+import ipaddress
+import warnings
+
+from django.conf import settings
+
+
+def _get_meta_ip_strs(request, header_name):
+    """
+    Get a list of IPs from a header in the given request.
+
+    Return the list of IPs the request is carrying on this header, which is
+    expected to be comma-delimited if it contains more than one. Response
+    may be an empty list for missing or empty header. List items may not be
+    valid IPs.
+    """
+    if not header_name:
+        return []
+
+    field_name = 'HTTP_' + header_name.replace('-', '_').upper()
+    header_value = request.META.get(field_name, '').strip()
```

Comment:
I'm not sure why I'm not using `request.headers` here, and instead seem to have duplicated the logic that that uses. I wonder if it's possible that this was to work around the "request" objects we get sometimes being `django.http.request.HttpRequest` and sometimes being a DRF wrapper? But the DRF wrapper does proxy through any "missing" attributes via a dynamic `__getattr__`, so `headers` is available. Maybe I thought it wasn't.

Anyway, I'm going to see if I can just replace this code with `request.headers.get(field_name, '').strip()`.